### PR TITLE
[lldb] Various changes to get building on `next`

### PR DESCRIPTION
**Pull request by bnbarham**

### CODE REVIEW

**Comment by bnbarham** [comment]
@JDevlieghere / @adrian-prantl how major is actually implementing the new methods in TypeSystemSwift/DWARFASTParserSwift?

File: lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp [hunk 1]
```diff
@@ -1349,7 +1349,7 @@ SwiftExpressionParser::ParseAndImport(
   if (enable_bare_slash_regex_literals) {
     lang_opts.enableFeature(swift::Feature::BareSlashRegexLiterals);
   }
-  if (uint32_t version = m_expr.Language().version) {
+  if (uint32_t version = m_expr.Language().version)
```
Comment (adrian-prantl) [inline lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp:1352]: This change looks scary.

Comment (adrian-prantl) [reply lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp:1352]: But I think it's correct.

**Review by adrian-prantl — APPROVED** [review]

File: lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp (hunk 1 above)
Comment (bnbarham) [reply lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp:1352]: Heh, yeah. I was going to add the `}` instead but this matches what was... somewhere (stable/20230725 I assume?)

**Pull request merged**